cache/fast_lru_cache.cc

```diff
@@ -368,27 +347,46 @@ Status LRUCacheShard::Insert(const Slice& key, uint32_t hash, void* value,
     EvictFromLRU(tmp.total_charge, &last_reference_list);
     if ((usage_ + tmp.total_charge > capacity_ &&
          (strict_capacity_limit_ || handle == nullptr)) ||
-        table_.GetOccupancy() == size_t{1} << table_.GetLengthBits()) {
-      // Originally, when strict_capacity_limit_ == false and handle != nullptr
-      // (i.e., the user wants to immediately get a reference to the new
-      // handle), the insertion would proceed even if the total charge already
-      // exceeds capacity. We can't do this now, because we can't physically
-      // insert a new handle when the table is at maximum occupancy.
+        table_.GetOccupancy() ==
+            static_cast<uint32_t>((uint32_t{1} << table_.GetLengthBits()) /
```

Comment:
We don't want to recompute this on every Insert(). Also, in case of miscalculation or potential lock-free future, `>=` seems safer than `==`

Reply:
I thought about ``>=`` in a future lock-free implementation. I used ``==`` because that's all we need in the current version.

Reply:
I'll add an assert to check we never go over the desired capacity.